### PR TITLE
clang-tidy: Fixes in passes 

### PR DESCRIPTION
**Pull request by MMory**

### CODE REVIEW

File: include/phasar/PhasarLLVM/Passes/ExampleModulePass.h [hunk 1]
```diff
@@ -21,7 +21,7 @@ namespace psr {
 class ExampleModulePass : public llvm::AnalysisInfoMixin<ExampleModulePass> {
 private:
   friend llvm::AnalysisInfoMixin<ExampleModulePass>;
-  static llvm::AnalysisKey Key;
+  static llvm::AnalysisKey Key; // NOLINT
```
Comment (vulder) [inline include/phasar/PhasarLLVM/Passes/ExampleModulePass.h:24]: Why does this static variable need a //NOLINT?

File: include/phasar/PhasarLLVM/Passes/GeneralStatisticsAnalysis.h [hunk 2]
```diff
@@ -138,14 +138,14 @@ class GeneralStatisticsAnalysis
     : public llvm::AnalysisInfoMixin<GeneralStatisticsAnalysis> {
 private:
   friend llvm::AnalysisInfoMixin<GeneralStatisticsAnalysis>;
-  static llvm::AnalysisKey Key;
+  static llvm::AnalysisKey Key; // NOLINT
```
Comment (vulder) [inline include/phasar/PhasarLLVM/Passes/GeneralStatisticsAnalysis.h:141]: same here

File: include/phasar/PhasarPass/Options.h [hunk 3]
```diff
@@ -15,25 +15,25 @@
 
 #include "llvm/Support/CommandLine.h"
 
-extern llvm::cl::OptionCategory PhASARCategory;
+extern llvm::cl::OptionCategory PhASARCategory; // NOLINT
```
Comment (vulder) [inline include/phasar/PhasarPass/Options.h:18]: Are externs not allowed? If this should be so, maybe we should consider removing the clang-tidy rule?

File: lib/PhasarPass/RegisterPasses.cpp [hunk 4]
```diff
@@ -16,57 +16,57 @@
...
+static llvm::cl::opt<std::string, true> SetPammOutputFile( // NOLINT
     "pamm-out", llvm::cl::desc("Filename for PAMM's gathered data"),
     llvm::cl::location(PammOutputFile), llvm::cl::init("PAMM_data.json"),
     llvm::cl::cat(PhASARCategory));
 
-bool psr::PrintEdgeRecorder;
+bool psr::PrintEdgeRecorder; // NOLINT
 static llvm::cl::opt<bool, true>
-    SetPrintEdgeRecorder("printedgerec",
+    SetPrintEdgeRecorder("printedgerec", // NOLINT
```
Comment (vulder) [inline lib/PhasarPass/RegisterPasses.cpp:55]: What rule is triggering here? This one works without errors 4 me.

**Review by vulder — CHANGES_REQUESTED** [review]

**Review by pdschubert — APPROVED** [review]

**Pull request merged**